### PR TITLE
fix: close mobile menu when clicked outside 

### LINKED ISSUE #3761: Mobile nav menu should close when clicked outside of it
### Description

In mobile view, user should be able to close the nav menu when clicked anywhere outside of it. 

### Actual Behavior

https://user-images.githubusercontent.com/116095538/218790455-5b5a0ed4-c2bb-4569-9b5d-363b4337f35d.mp4

### Expected Behavior

https://user-images.githubusercontent.com/116095538/218790232-91e55638-90b1-44a4-9fff-429c8485bc84.mp4

### Environment:
- Host OS: Linux
- Browser: Chrome

---
<h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
- 🎨 Wireframes and designs for Layer5 site in [Figma](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). 
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io).

Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

### PR DESCRIPTION
**Pull request by swagataroy30**
**Description**

This PR fixes #3761 

**Notes for Reviewers**

- Close mobile menu when clicked outside
- Close mobile menu when clicked on `Layer5` logo

### Before
https://user-images.githubusercontent.com/116095538/219590314-d9dc2247-bab6-45b1-b61d-e2c46c5217c0.mp4

### After
https://user-images.githubusercontent.com/116095538/219590399-18e71294-1dd3-433d-9d9f-7c4e8c427ce5.mp4

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit c1d94c60deff7f65b32970ab055c95902cd1346a at: https://63ef566980ffae5ba12e8f18--layer5.netlify.app

File: src/sections/General/Navigation/index.js [hunk 1]
```diff
@@ -207,11 +221,14 @@ const Navigation = ({ theme, themeSetter }) => {
...
-      <NavigationWrap className={`nav-block ${scroll ? "scrolled" : ""}`}>
+      <NavigationWrap className={`nav-block ${scroll ? "scrolled" : ""}`} ref={navWrapRef}>
         <Container className="nav-container">
           <div className="navbar-wrap">
             <Link to="/" className="logo">
-              <img src={theme === "dark" ? layer5dark_logo : layer5_logo} alt="Layer5 logo" />
+              <img src={theme === "dark" ? layer5dark_logo : layer5_logo} alt="Layer5 logo"
+                onClick={function () {
+                  setExpand(false); closeDropDown();
+                }}/>
```
Comment (Nikhil-Ladha) [inline src/sections/General/Navigation/index.js:231]: Let's not add it to the logo, just having an event listener for outside the header is fine.

Comment (swagataroy30) [reply src/sections/General/Navigation/index.js:231]: Okay! Thanks for the reply. I will make the said changes.

**Comment by l5io** [comment]
🚀 Preview for commit a2a4d48f0d58df1d653d33149ff328801cb9e26e at: https://63f2600c6621d30ee1c052ca--layer5.netlify.app

**Review by Nikhil-Ladha — APPROVED** [review]

**Pull request merged**